src/Cake.Common/Tools/ILMerge/TargetPlatform.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+
+namespace Cake.Common.Tools.ILMerge
+{
+    /// <summary>
+    ///     Represents target platform option
+    ///     Command line option: /targetplatform:version,platformdirectory
+    /// </summary>
+    /// <example> /targetPlatform:"v4,C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.1" </example>
+    public class TargetPlatform
+    {
+        /// <summary>
+        ///     Initializes a new instance of the TargetPlatform class.
+        /// </summary>
+        /// <param name="platform"> .NET Framework target version </param>
+        /// <param name="dir"> This is a directory where mscorlib.dll can be found</param>
+        public TargetPlatform(TargetPlatformVersion platform, string dir)
+        {
+            Platform = platform;
```

Comment:
* `dir` should be `Cake.Core.IO.DirectoryPath` instead of string.
* Rename `dir` to path.
* Validate directory against null and throw `ArgumentNullException`.